### ISSUE TITLE
Saved searches [improvement]

### ISSUE THREAD

**Issue by yannouchou** [janeczku/calibre-web#1129]
I would like to suggest a new feature to let a user save a search and add it to the left panel (similarly to shelves).
The advantage of these saved dynamic searches over the manually-managed shelves is that the searches can be based on standard Calibre fields such as tags, so they could easily be synced directly in the Calibre UI (and its equivalent concepts of saved searches and virtual libraries).

For instance, I use Calibre tags to manage "to-read", "read", "favorite" lists directly in Calibre, it would be great to have those easily accessible and synced in Calibre-web...

Many thanks!

**Comment by EolianPipes** [comment]
This would be awesome! +1

I've been using Calibre and Calibre-Web to store audiobooks and have a checkbox field in Calibre to keep track of which books have an added audiobook. I know that I can do an "advanced search" for that specific field, but it was a bit cumbersome to explain that to my users who aren't as tech savvy. I've been playing around with bulk adding a search to a shelf, but it's hard to keep it updated as things are being added.

**Comment by fivestones** [comment]
I'd love to have this feature too +1

**Comment by fivestones** [comment]
Another option that could maybe be a workaround is if you could send a search as a get request. That way you could just save the url to a search result in bookmarks, and when you loaded that bookmark it would be loading the saved search. I would definitely prefer the ability to have a built-in saved search feature though.

**Comment by mensoh** [comment]
I too would love this feature, perhaps as 'Dynamic shelf' ?
I would love to have a shelf containing all 'regular' fiction in the library excluding fantasy and sci-fi for example. Would be happy to see if I can implement it or do you have ideas for this already @OzzieIsaacs ?

**Comment by mendhak** [comment]
I use Calibre Web's Kobo sync feature. Having a dynamic shelf for 'unread' would be useful, it would be amazing to have just my unread books synced to the device, and new appear as I add them to Calibre.

**Comment by Protarios** [comment]
I would prefer to be able to have the ability to more flexible link to custom columns in Calibre.
This would allow to sync shelves from Calibre-Web to a custom column, that can be used as shelves in Calibre.

Idealy also Calibre Virtual Libraries could be supported, as they are a great way to filter complex subsets of a large library.

**Comment by Humam-Hamdan** [comment]
Yep, I use Virtual Libraries and Custom columns (VLs for Broad Tags and CCs for Read Status), while i can get what i want using advanced search, it'd be amazing if it can be added as a shelf.
